kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java

```diff
@@ -346,7 +414,14 @@ private String sanitizeTableName(String tableName) {
    *         name if sanitization is enabled.
    */
   public String getTableFromTopic(String topic) {
-    return getBoolean(SANITIZE_TOPICS_CONFIG) ? sanitizeTableName(topic) : topic;
+    String tableName = null;
+    Map<String, String> map = getTopicsToTables();
```

Comment:
OK, I see the issue. My thought: move this stuff into BigQuerySinkConnector. The sink connector creates all of the tasks. I think the SinkTask should just take a static map from topic -> table. It shouldn't deal with doing any of the regex/resolving stuff. It should just do O(1) lookups on its map for each msg to get the table that a given msg is going into.

The SinkConnector can do all of the topic->table resolution one time, and then pass the final maps into the sink task. I *think* that should work, and will be a bit cleaner than dumping a bunch of utility logic into the config class (which is what we were doing even before you sent in this PR).